### PR TITLE
fix(preProcessPatterns): support glob context paths with special characters

### PR DESCRIPTION
**Pull request by clydin**
`context` paths with special characters (i.e., glob characters) will cause files to not be copied.  This is especially important when the context represents the absolute path to a location and the path is not under the control of the project using the plugin.

Tests have been updated and additional test cases added.

Addresses: https://github.com/angular/angular-cli/issues/9230

### CODE REVIEW

**Comment by clydin** [comment]
Oddly, `apply function options ignore ignores all files except those with dots` fails on node 4/6 but passes on 8.  Will investigate.

**Comment by clydin** [comment]
Alright.  Update to be more comprehensive and with additional test cases.

File: src/preProcessPattern.js [hunk 1]
```diff
@@ -7,6 +7,29 @@ import isObject from './utils/isObject';
 // https://www.debuggex.com/r/VH2yS2mvJOitiyr3
 const isTemplateLike = /(\[ext\])|(\[name\])|(\[path\])|(\[folder\])|(\[emoji(:\d+)?\])|(\[(\w+:)?hash(:\w+)?(:\d+)?\])|(\[\d+\])/;
 
+function escape(context, from) {
```
Comment (evilebottnawi) [inline src/preProcessPattern.js]: Can you move `escape` to `src/utils`

Comment (clydin) [reply src/preProcessPattern.js]: Done.

**Pull request merged**